File: Src/java/elm-fhir/src/main/java/org/cqframework/cql/elm/evaluation/TestLibrarySourceProvider.java
package org.cqframework.cql.elm.evaluation;

import kotlinx.io.Source;
import org.cqframework.cql.cql2elm.LibraryContentType;
import org.cqframework.cql.cql2elm.LibrarySourceProvider;
import org.hl7.elm.r1.VersionedIdentifier;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;

public class TestLibrarySourceProvider implements LibrarySourceProvider {
    @Override
    public Source getLibrarySource(VersionedIdentifier libraryIdentifier) {
        String libraryFileName = String.format(
                "%s.cql",
                libraryIdentifier
                        .getId()); // , libraryIdentifier.getVersion() != null ? ("-" + libraryIdentifier.getVersion())
        // : "");
        var inputStream = TestLibrarySourceProvider.class.getResourceAsStream(libraryFileName);
        return inputStream == null ? null : buffered(asSource(inputStream));
    }

    @Override
    public Source getLibraryContent(VersionedIdentifier libraryIdentifier, LibraryContentType type) {
        if (LibraryContentType.CQL == type) {
            return getLibrarySource(libraryIdentifier);
        }

        return null;
    }
}

File: Src/java/elm-fhir/src/main/java/org/cqframework/cql/elm/evaluation/TestLibrarySourceProvider.kt
package org.cqframework.cql.elm.evaluation

import kotlinx.io.Source
import kotlinx.io.asSource
import kotlinx.io.buffered
import org.cqframework.cql.cql2elm.LibraryContentType
import org.cqframework.cql.cql2elm.LibrarySourceProvider
import org.hl7.elm.r1.VersionedIdentifier

class TestLibrarySourceProvider : LibrarySourceProvider {
    override fun getLibrarySource(libraryIdentifier: VersionedIdentifier): Source? {
        val libraryFileName =
            String.format(
                "%s.cql",
                libraryIdentifier.id,
            ) // , libraryIdentifier.getVersion() != null ? ("-" + libraryIdentifier.getVersion())
        // : "");
        val inputStream = TestLibrarySourceProvider::class.java.getResourceAsStream(libraryFileName)
        return if (inputStream == null) null else inputStream.asSource().buffered()
    }

    override fun getLibraryContent(
        libraryIdentifier: VersionedIdentifier,
        type: LibraryContentType,
    ): Source? {
        if (LibraryContentType.CQL == type) {
            return getLibrarySource(libraryIdentifier)
        }

        return null
    }
}
